src/Parser/Warrior/Arms/Modules/Features/ExecuteRange.js

```diff
@@ -0,0 +1,69 @@
+import Analyzer from 'Parser/Core/Analyzer';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+
+const EXECUTE_RANGE = 0.2;
+
+/**
+ * Tracks whether enemies are in Execute range through damage events so that it can be accessed in cast events by other modules.
+ * @extends Analyzer
+ */
+class ExecuteRangeTracker extends Analyzer {
+  enemyMap = {};
+
+  on_byPlayer_damage(event) {
+    if(event.targetIsFriendly) {
+      return;
+    }
+
+    const targetString = encodeTargetString(event.targetID, event.targetInstance);
+
+    if(this.enemyMap[targetString] === undefined) {
+      this.enemyMap[targetString] = [];
+    }
+
+    const inExecuteRange = event.hitPoints / event.maxHitPoints <= EXECUTE_RANGE;
+
+    const timeline = this.enemyMap[targetString];
+    const newEvent = timeline.length === 0 || timeline[timeline.length - 1].timestamp !== event.timestamp;
+    const differentState = timeline.length === 0 || timeline[timeline.length - 1].inExecuteRange !== inExecuteRange;
+
+    if(!differentState) {
+      // Stop if the state is the same as the last event.
+      return;
+    }
+
+    if(newEvent) {
+      // If this event is more recent than the last event, record it.
+      timeline.push({
+        timestamp: event.timestamp,
+        inExecuteRange: inExecuteRange,
+      });
+    } else if(!newEvent) {
```

Comment:
the if statement is redundant here since you already have the else from the above if statement.